index.js

```diff
@@ -188,6 +190,10 @@ function parse(input, options) {
 		// http://w3.org/TR/2012/WD-url-20120524/#collect-url-parameters
 		value = value === undefined ? null : decode(value, options);
 
+		if (options.number && isNumber(value)) {
```

Comment:
You don't need a module for this. Just try `Number(value)` and see if it's NaN or not.